src/hotspot/share/memory/virtualspace.cpp

```diff
@@ -153,6 +207,49 @@ void ReservedSpace::initialize_members(char* base, size_t size, size_t alignment
   _noaccess_prefix = 0;
 }
 
+void ReservedSpace::reserve(size_t size, size_t alignment, bool large,
+                            char* requested_address,
+                            bool executable) {
+  assert(is_aligned(size, alignment), "Size must be aligned to the requested alignment");
+
+  // There are basically three different cases that we need to handle below:
+  // - Mapping backed by a file
+  // - Mapping backed by explicit large pages
```

Comment:
I think there should be a description what "explicit large pages" are. Only in the case that the OS can't commit large page memory this path is taken.
The third option kind of suggests that there is only "explicit large page support" if you want large pages (only talking about normal pages or THP, but on e.g. Linux you do not need to pre-commit large pages reservations.

Reply:
@kstefanj enlightened me about that even on Linux we pre-commit everything *but* the linux counters (particularly `free_hugepages`) is not updated until the page has been touched. That misled me to believing that we do not need to actually precommit there.
So, ignore this comment.

Reply:
Discussed this with Thomas offline and we agreed on the current text being ok after all.